### PR TITLE
Add large data to Advanced config example

### PR DESCRIPTION
**Pull request by elianalf**
## Description

This pr adds the possibility to send large data with the advanced configuration example. The user can define the size of the message that is sent.  

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: examples/cpp/dds/AdvancedConfigurationExample/AdvancedConfigurationPublisher.cpp [hunk 1]
```diff
@@ -176,7 +179,7 @@ bool HelloWorldPublisher::init(
     }
 
     // CREATE THE TOPIC
-    topic_ = participant_->create_topic(topic_name, "HelloWorld", TOPIC_QOS_DEFAULT);
+    topic_ = participant_->create_topic(topic_name, "LargeData", TOPIC_QOS_DEFAULT);
```
Comment (rsanchez15) [inline examples/cpp/dds/AdvancedConfigurationExample/AdvancedConfigurationPublisher.cpp:182]:
```suggestion
    topic_ = participant_->create_topic(topic_name, type_.get_type_name(), TOPIC_QOS_DEFAULT);
```

File: examples/cpp/dds/AdvancedConfigurationExample/AdvancedConfigurationPublisher.cpp [hunk 2]
```diff
@@ -325,8 +328,9 @@ void HelloWorldPublisher::runThread(
         if (listener_.enough_matched())
         {
             publish();
-            std::cout << "Message: " << hello_.message().data() << " with index: " << hello_.index()
-                      << " SENT" << std::endl;
+            std::cout << "Message: " << hello_.message().data() << " with index: " << hello_.index() <<
+                "  and data size: " << static_cast<int>(hello_.data().size())
+                      << "  SENT" << std::endl;
```
Comment (rsanchez15) [inline examples/cpp/dds/AdvancedConfigurationExample/AdvancedConfigurationPublisher.cpp:333]:
```suggestion
            std::cout << "Sample sent: " << hello_.message().data() << "  " << hello_.index() <<
                "  (" << static_cast<int>(hello_.data().size()) << " Bytes)"
                      << std::endl;
```

File: examples/cpp/dds/AdvancedConfigurationExample/AdvancedConfigurationSubscriber.cpp [hunk 3]
```diff
@@ -288,7 +288,9 @@ void HelloWorldSubscriber::SubListener::on_data_available(
         {
             samples_++;
             // Print your structure data here.
-            std::cout << "Message " << hello_.message().data() << " " << hello_.index() << " RECEIVED" << std::endl;
+            std::cout << "Message " << hello_.message().data() << " " << hello_.index() << "  Data size " <<
+                static_cast<int>(hello_.data().size())  << "  RECEIVED" << std::endl;
```
Comment (rsanchez15) [inline examples/cpp/dds/AdvancedConfigurationExample/AdvancedConfigurationSubscriber.cpp:292]:
```suggestion
            std::cout << "Sample received: " << hello_.message().data() << "  " << hello_.index() <<
                "  (" << static_cast<int>(hello_.data().size()) << " Bytes)"  << std::endl;
```

File: examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h [hunk 4]
```diff
@@ -186,12 +187,14 @@ const option::Descriptor usage[] = {
 
     { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nPublisher options:"},
     { TOPIC, 0, "t", "topic",                  Arg::String,
-      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)." },
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: LargeDataTopic)." },
```
Comment (rsanchez15) [inline examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h:190]:
```suggestion
      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: AdvancedConfigurationTopic)." },
```

File: examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h [hunk 5]
```diff
@@ -186,12 +187,14 @@ const option::Descriptor usage[] = {
...
     { DOMAIN_ID, 0, "d", "domain",                Arg::NumericRange<0, 230>,
       "  -d <id> \t--domain=<id>  \tDDS domain ID (Default: 0)." },
     { WAIT, 0, "w", "wait",                 Arg::NumericRange<>,
       "  -w <num> \t--wait=<num> \tNumber of matched subscribers required to publish"
       " (Default: 0 => does not wait)." },
+    { DATA_SIZE, 0, "m", "msg-size",         Arg::String,
+      "  -m <num> \t--msg-size=<num>\tNumber of bytes sent as data in the message (Default 10)."},
```
Comment (rsanchez15) [inline examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h:197]:
```suggestion
      "  -m <num> \t--msg-size=<num>\tSize in bytes of the data to be sent (Default 10)."},
```

File: examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h [hunk 6]
```diff
@@ -209,7 +212,7 @@ const option::Descriptor usage[] = {
 
     { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nSubscriber options:"},
     { TOPIC, 0, "t", "topic",                  Arg::String,
-      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)." },
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: LargeDataTopic)." },
```
Comment (rsanchez15) [inline examples/cpp/dds/AdvancedConfigurationExample/arg_configuration.h:215]:
```suggestion
      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: AdvancedConfigurationTopic)." },
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Review by rsanchez15 — DISMISSED** [review]

**Review by EduPonz — CHANGES_REQUESTED** [review]
I think we should state in the readme of the example that due to the sequence in the data type this example will not use data sharing

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Comment by EduPonz** [comment]
@richiprosima please test this

**Comment by elianalf** [comment]
@richiprosima please test mac

**Comment by elianalf** [comment]
@richiprosima please test mac

**Pull request merged**